### DISCUSSION TITLE
Record `with` expression variables

### DISCUSSION THREAD

**Discussion by thomhurst** [dotnet/csharplang#7531] [Language Ideas]
Complex models with many nested properties require a lot of chaining in order to override properties deep within the structure. 

An example could be:

```csharp
var myData = _fixture.Create<MyData>()
myData = myData with
{
    SomeProperty = myData.SomeProperty with
    {
        SomeNestedProperty = myData.SomeProperty.SomeNestedProperty with
        {
                Nested2 = myData.SomeProperty.SomeNestedProperty.Nested2 with
                {
                        Nested3 = myData.SomeProperty.SomeNestedProperty.Nested2.Nested3 with 
                        {
                                SomeInteger = 7;
                        }
                }
        }
    }
};
```

Regardless of how deep in you are, you always need to start by referencing the top level variable (in this case, `myData`), unless you start creating more variables separately outside of the `with` expression.

An idea is optionally creating a temporary variable within some parenthesizes when using the `with` keyword.  This would provide a reference to the record that is currently being cloned, so you can reference it more easily.

This can reduce the need for long call chains in complex models.

```csharp
var myData = _fixture.Create<MyData>() with(myData)
{
    SomeProperty = myData.SomeProperty with(someProperty) 
    {
        SomeNestedProperty = someProperty.SomeNestedProperty with(someNestedProperty)
        {
                Nested2 = someNestedProperty.Nested2 with(nested2) 
                {
                        Nested3 = nested2.Nested3 with 
                        {
                                SomeInteger = 7;
                        }
                }
        }
    }
};
```

**Comment by HaloFour** [comment]
See: #77

It's a little different but intends to solve a similar problem by allowing the "wither" syntax to modify nested property paths, e.g.:

```cs
var myData = _fixture.Create<MyData>() with { SomeProperty.SomeNestedProperty.Nested2.Nested3.SomeInteger = 7 };
```
